pieforproviders/pieforproviders
# [PIE-913] Updates the attendance processor and associated logic

Updates the attendance processor and associated logic; removes null constraint from check out time, updates the time param validator to use default error handling, adds missing field spec runs for dashboard and attendance, adds wonderschool ID to attendances, adds safe operators to active_child_approval, removes 0 validation on total_time_in_care

# 💅🏼 What issue does this fix?

#913 

## 🍂 Before You Submit


## 🛷 Deployment


## 🧵 Steps to set up locally

```
bundle exec rails db:migrate
```

## 🧳 Steps to test
correct records:
- create two S3 buckets (one for uploading, one for archiving) if you haven't already and configure your env. vars to use those buckets for the attendance s3 bucket and archive
- put a correctly formatted attendance csv in the uploading S3 bucket (see `spec/fixtures/files/wonderschool_necc_attendance_data.csv` for example) with records for children in your db - the child should have a `wonderschool_id` that matches the `child_id` in the spreadsheet, and belong to a business where the `name` matches the `school_name` in the spreadsheet
- run `bundle exec rails read_wonderschool_necc_attendances`
- [ ] your attendances should be created correctly
- [ ] your uploaded CSV should be archived to the other bucket

invalid format:
- add an invalid format csv to the uploading S3 bucket (see `spec/fixtures/files/wonderschool_necc_attendance_data_invalid_format.csv` for example)
- run `bundle exec rails read_wonderschool_necc_attendances`
- [ ] a `failed_records` csv should be uploaded to the archive s3
- [ ] your original csv should not be archived

missing field:
- add a csv that is missing a required field to the uploading S3 bucket (see `spec/fixtures/files/wonderschool_necc_attendance_data_missing_field.csv` for example)
- run `bundle exec rails read_wonderschool_necc_attendances`
- [ ] a `failed_records` csv should be uploaded to the archive s3
- [ ] your original csv should not be archived

## 🕯 Caveats, concerns
This will have merge conflicts w/ @wittejm 's ongoing PR but I can handle those after his is merged
